lib/intercom-rails/auto_include_filter.rb

```diff
@@ -56,7 +65,16 @@ def intercom_script_tag_called_manually?
       end
 
       def intercom_script_tag
-        @script_tag ||= ScriptTag.new(:find_current_user_details => true, :find_current_company_details => true, :controller => controller, :show_everywhere => show_everywhere?)
+        # User defined method for applying a nonce to the inserted js tag when
+        # using CSP
+        if defined?(CoreExtensions::IntercomRails::AutoInclude.csp_nonce_hook) == 'method'
```

Comment:
I feel like this bit could be DRYd up a bit, give me a few minutes...

Reply:
I know it's nearly two years later, but I just hit a case where _not_ memoizing the `@script_tag` is having an impact. Specially, it means the `ScriptTag` ends up calling `ScriptTag#find_current_user_details` twice (b/c we initialize the `script_tag` twice in the `AutoIncludeFilter::filter` method). That results in running all `Proc`s used to configure the current user, customer, and any details. Which means we're doing some work twice. And some of that work might be DB calls.

So... any reason _not_ to memoize here?